src/main/java/sparksoniq/jsoniq/runtime/iterator/operational/UnaryOperationIterator.java

```diff
@@ -48,9 +48,13 @@ public Item next() {
         if (this._hasNext) {
             this._hasNext = false;
             _child.open(_currentDynamicContext);
-            Item child = _child.next();
+            Item child = null;
+            if (_child.hasNext()) {
+                child = _child.next();
+            }
             _child.close();
-
+            if (child == null)
+                return null;
```

Comment:
One small but important thing here: next() should never return null, but throw an error in unexpected cases. Under normal operation, next() must return a non-null Item if hasNext() returned true, and it should not have been invoked in the first place if hasNext() returned false (e.g., for empty sequence or after the last item has been consumed).

If null is returned rather than throwing an IteratorFlowException, this may "hide" any errors in the code under the carpet, leading to bugs later on.

Reply:
Thank you for spotting this. 
From revision 3, i.e. commit 16e34f9, I reverted the changes that I previously made to UnaryOperationIterator and now InstanceOfIterator extends UnaryOperationBaseIterator so that calling open() on an InstanceOfIterator object will use RuntimeIterator open() function and not the one overriding it in UnaryOperationIterator.

Reply:
As for creating branches from the main rumbleDB repository, sure, I will do. I just forked because initally I was not Contributor to the main repo. 

As soon as you approve the most recent changes, I will proceed with the merge.